### DISCUSSION TITLE
Dealing with failed OTA device updates

### DISCUSSION THREAD

**Discussion by benflux** [Koenkk/zigbee2mqtt#17697] [General]
How do I cancel a failed OTA device update that is stuck and not progressing?

![image](https://github.com/Koenkk/zigbee2mqtt/assets/31230445/5be68204-4faa-44b3-ba87-e465611360ec)

**Comment by benblazak** [comment]
same question, can't find anything about it in the docs or google

**Comment by benblazak** [comment]
https://github.com/Koenkk/zigbee2mqtt/issues/10666#issuecomment-1008362778

looks like you can stop z2m and clear the state manually in data/state.json

**Comment by sjorge** [comment]
You can also trigger a 'check' via mqtt which will reset the state.

```
mosquitto_pub -t zigbee2mqtt/bridge/request/device/ota_update/check -m '{"id": "light/kitchen/table/bulb1"}'
```

**Comment by diphen** [comment]
I don't know how to manually publish via MQTT, but if you're using the MQTT broker you can do it via the UI:

- Settings -> Devices & Services -> MQTT -> Configure -> Publish a packet:
      Topic:         zigbee2mqtt/bridge/request/device/ota_update/check
      Payload:    {"id": "deviceID"}
                Where "deviceID" is either the device's IEEE address, or the friendly name, like what sjorge used above.
- Then click "PUBLISH"
      You won't see the UI do anything, but when you go back to the z2m OTA page, you'll see the devices are no longer stuck.


Personally I find the IEEE address easier since you can copy/paste it straight from the z2m UI.  For example my publish looked like this:

<img width="470" alt="image" src="https://github.com/Koenkk/zigbee2mqtt/assets/563825/89e9252e-5ef5-47cf-8f2e-4e1b43245df5">
